typespec-extension/src/code-model-builder.ts

```diff
@@ -543,13 +543,20 @@ export class CodeModelBuilder {
     this.processRouteForLongRunning(codeModelOperation, op.responses, lroMetadata);
 
     operationGroup.addOperation(codeModelOperation);
-    
+
     return codeModelOperation;
   }
 
   private isMultipleContentTypes(httpOperation: HttpOperation): boolean {
     if (httpOperation.parameters.parameters && httpOperation.parameters.parameters.length > 0) {
-      if (httpOperation.parameters.parameters.some(parameter => parameter?.type === "header" && parameter?.name === "content-type" && parameter?.param?.type?.kind === "Union")) {
+      if (
+        httpOperation.parameters.parameters.some(
+          (parameter) =>
+            parameter?.type === "header" &&
+            parameter?.name === "content-type" &&
```

Comment:
Do you need `toLowerCase()`?

Reply:
https://github.com/microsoft/typespec/blob/main/packages/http/src/content-types.ts#L9 - it's case sensitive.

No matter using "contentType" or "ContentType" in tsp, the "parameter.name" here is `content-type`.

[playground](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZ8wfYXp1cmUtdG9vbHMvyCstcHJvdmlkZXJodWLfLGMtxhVjb3Jl3yvIK3Jlc291cmNlLW1hbmFnZXIiOwoKdXNpbmcgVHlwZVNwZWMuSHR0cDvQFVJlc3TRFVbpANXIG0HEZS5Db3JlzhJSx3dNxnY7CgpAYXJtUOcA6k5hbWVzcGFjZQpAc2VydmljZSh7CiAgdGl0bGU6ICJDb250b3NvyC5IdWJDbGllbnQiLAogIOcBWTogIjIwMjEtMTAtMDEtcHJldmlldyIsCn0pCkBkb2MoyEUg6ACKIMhPIOYBCm1lbnQgQVBJLiIpCkB1c2VEZXBlbmRlbmN5KPUAw%2BgA%2FXMudjFfMF9QxnJfMSkKbugAziBNaWNyb3NvZnQu8gDD5AED5QCZbXVsdGlwbGUgZGF0YSDlAa8gbWFwIHRvIMkbY29udOQAoMUe5AChcm91dGUoInVwbG9hZC9vdmVyxQnFMS1ib2R5LcUrMcQscG9zdApvcCDGLVdpdGhPxzAxKAogIEDEL8V9OiBzdHLkAbR8IGJ5dGVzIHzpAS7kAWpAaGVh5AEw5ADMZW505AIGOiAgInRleHQvcGxhaW4iIHwgImFwcGxpY2F0aW9uL2pzb9IVb2N0ZXQtc3RyZWFtxR1pbWFnZS9qcGVnyw9wbmciCik6IHZvaWT%2FASv%2FASv%2FASvoASsy%2FgErMv8BK%2FYBK%2BcAjf8BK%2F8BK%2F8BK%2B8BKw%3D%3D)

Reply:
Not "contentType" or "ContentType"

The `@header("Content-Type")`. They are different thing, please...

Reply:
```
@header("Content-Type") WhateverName:  "text/plain" | "application/json" | "application/octet-stream" | "image/jpeg" | "image/png"
```

Reply:
Sorry, my fault. it's case insensitive, https://github.com/microsoft/typespec/blob/main/packages/http/src/content-types.ts#L13

Using `@header("Content-Type")` also works, and parameter.name is `Content-Type`. I will update the code.